### PR TITLE
Polish rendering driver refactor further (take 2)

### PR DESCRIPTION
**Pull request by RandomShaper**
Followup of #65541 (in turn, a small contribution to the #65541 endeavor).

- Keep replacing terms such as video driver or rendering backend to 'rendering driver'.
- Rename `vulkan_color` to `highend_color`. Additional work will happen in the topic of what highend exactly means in the different parts of the editor (visual shaders, general UI), but getting rid of the fact that a specific driver has its own color (as opposed to a rendering method) is already a step forward.

### CODE REVIEW

File: editor/plugins/node_3d_editor_plugin.cpp [hunk 1]
```diff
@@ -4778,7 +4778,7 @@ Node3DEditorViewport::Node3DEditorViewport(Node3DEditor *p_spatial_editor, int p
 	view_menu->set_disable_shortcuts(true);
 
 	// TODO: Re-evaluate with new OpenGL3 renderer, and implement.
-	//if (OS::get_singleton()->get_current_video_driver() == OS::VIDEO_DRIVER_GLES2) {
+	//if (OS::get_singleton()->get_current_video_driver() == OS::RENDERING_DRIVER_GLES3) {
```
Comment (akien-mga) [inline editor/plugins/node_3d_editor_plugin.cpp:4781]: It's `RENDERING_DRIVER_OPENGL_3`.

File: core/core_bind.h [hunk 2]
```diff
@@ -124,9 +124,9 @@ class OS : public Object {
 	static OS *singleton;
 
 public:
-	enum VideoDriver {
-		VIDEO_DRIVER_VULKAN,
-		VIDEO_DRIVER_OPENGL_3,
+	enum RenderingDriver {
+		RENDERING_DRIVER_VULKAN,
+		RENDERING_DRIVER_OPENGL_3,
```
Comment (akien-mga) [inline core/core_bind.h:129]: Personally I think it would look better as `_OPENGL3` instead of `_OPENGL_3`, but I'm fine with either :)

Comment (RandomShaper) [reply core/core_bind.h:129]: Me too, to be honest. I keep seeing a double underscore (`OPENGL__3`) all the time. Would `GLES3` be an option or have we departed from the `ES` naming?

Comment (clayjohn) [reply core/core_bind.h:129]: I would go with ``RENDERING_DRIVER_OPENGL3`` We are trying to consistently use ``OpenGL3`` for user-facing areas and GLES3 internally.

File: platform/uwp/os_uwp.cpp [hunk 3]
```diff
@@ -159,7 +159,7 @@ Error OS_UWP::initialize(const VideoMode &p_desired, int p_video_driver, int p_a
 	outside = true;
 
 	// FIXME: Hardcoded for now, add Vulkan support.
-	p_video_driver = VIDEO_DRIVER_OPENGL;
+	p_video_driver = RENDERING_DRIVER_OPENGL;
```
Comment (clayjohn) [inline platform/uwp/os_uwp.cpp:162]:
```suggestion
	p_video_driver = RENDERING_DRIVER_OPENGL3;
```

**Review by clayjohn — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**